lib/logstash/inputs/s3.rb

```diff
@@ -253,18 +261,15 @@ def read_plain_file(filename, block)
 
   private
   def read_gzip_file(filename, block)
-    # Details about multiple streams and the usage of unused from: http://code.activestate.com/lists/ruby-talk/11168/
-    File.open(filename) do |zio|
-      while true do
-        io = Zlib::GzipReader.new(zio)
-        io.each_line { |line| block.call(line) }
-        unused = io.unused
-        io.finish
-        break if unused.nil?
-        zio.pos -= unused.length # reset the position to the other block in the stream
-      end
+    fileStream = FileInputStream.new(filename)
+    gzipStream = GZIPInputStream.new(fileStream)
+    decoder = InputStreamReader.new(gzipStream, "UTF-8")
```

Comment:
This presumes text encoding ? Should this be a ByteAraryInputStream ?

Reply:
The current implementation implicitly assumes UTF-8; we could make this configurable?